CONTRIBUTING.md

```diff
@@ -26,4 +26,16 @@ Here is how to contribute back some code or documentation:
 - Submit a pull request against the `dev` branch.
 - Be kind
 
+## How to add a function from OpenCV to GoCV
+
+Here are a few basic guidelines on how to add a function from OpenCV to GoCV:
+
+- Please open a Github issue. We want to help, and also make sure that there is no duplications of efforts. Sometimes what you need is already being worked on by someone else.
+- Use the proper Go style naming `MissingFunction()` for the Go wrapper.
+- Make any output parameters `Mat*` to indicate to developers that the underlying OpenCV data will be changed by the function.
+- Use Go types when possible as parameters for example `image.Point` and then convert to the appropriate OpenCV struct. Also define a new type based on `int` and `const` values instead of just passing "magic numbers" as params. For example, the `VideoCaptureProperties` type used in `videoio.go`.
+- Always add the function to the GoCV file named the same as the OpenCV module to which the function belongs.
+- If the new function is in a module that is not yet implemented by GoCV, a new set of files for that module will need to be added.
+- Always add a "smoke" test for the new function being added. We are not testing OpenCV itself, but just the GoCV wrapper, so all that is needed generally is just exercising the new function.
```

Comment:
should we add a note about the `WithParams` functions?

Reply:
Done.